src/api/nodeRpcCall.js

```diff
@@ -1884,10 +1868,16 @@ const congressProposeRepealLegislation = async (
 
   const congressmen = await api.query.council.members();
   const threshold = Math.trunc(congressmen.length/2) + 1;
-  const extrinsic = api.tx.utility.batchAll([
-    api.tx.preimage.notePreimage(repealLaw.toHex()),
-    api.tx.council.propose(threshold, proposal, proposal.length),
-  ])
+
+  const existingPreimages = await api.query.preimage.preimageFor([repealLaw.hash, repealLaw.encodedLength])
```

Comment:
```suggestion
  const existingPreimage = await api.query.preimage.preimageFor([repealLaw.hash, repealLaw.encodedLength])
```
But TBH I would reduce it to just preimage